src/main/java/frc/robot/commands/LockMode/Lock.java

```diff
@@ -0,0 +1,54 @@
+package frc.robot.commands.LockMode;
+
+import java.util.function.DoubleSupplier;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Camera.Camera;
+import frc.robot.subsystems.drive.DriveSubsystem;
+
+
+
+public class Lock extends CommandBase{
+    //Subsystem requirements for the command
+    public final void addRequirements(Camera camera){}
+    public final void addRequirements(DriveSubsystem drive){}
+
+    DriveSubsystem drive;
+    DoubleSupplier sideways;
+    DoubleSupplier forward;
+    boolean lockedOn = false;
+    Camera camera;
+
+    //PID controller for yawRate
+    PIDController yawRateController = new PIDController(1, 0, 0);
+
+    //This gets the controller inputs and drive information    
+    public Lock(DriveSubsystem drive, DoubleSupplier forward, DoubleSupplier sideways){
+        this.drive = drive;
+        this.forward = forward;
+        this.sideways = sideways;
+    }
+
+    public void cam(Camera camera){
+        this.camera = camera;
+        
+    }
+    
+    
+    
+    @Override
+    public void initialize(){}  
```

Comment:
If you are using Integral gain ( $kI \neq 0$ ) and to some degree if you are using derivative gain, the PID controller relies on past measurements that it stores. If you stop for a while and do something else, those past measurements need to be invalidated by calling `pid_controller.reset()`

Reply:
Would it be placed in the execute method?
`public  void execute(){
        double angularOffset = Camera.getAngle();
        double yawRate = yawRateController.calculate(angularOffset, 0);
        drive.drive(forward.getAsDouble(), sideways.getAsDouble(), yawRate, true);
        yawRateController.reset();
    }`

Reply:
In execute it would reset every calculation. The point of integral is to carry information between consecutive calculations, so we wouldn't want it in reset then. It's when the command is first being scheduled that we want to drop previous calculations, which are outdated, so `yawRateController.reset();` should be in `initialize();`